pkg/api/validation/validation.go

```diff
@@ -570,23 +573,36 @@ func validateDownwardAPIVolumeSource(downwardAPIVolume *api.DownwardAPIVolumeSou
 		if len(downwardAPIVolumeFile.Path) == 0 {
 			allErrs = append(allErrs, validation.NewRequiredError("path"))
 		}
-		if path.IsAbs(downwardAPIVolumeFile.Path) {
-			allErrs = append(allErrs, validation.NewForbiddenError("path", "must not be an absolute path"))
-		}
-		items := strings.Split(downwardAPIVolumeFile.Path, string(os.PathSeparator))
-		for _, item := range items {
-			if item == ".." {
-				allErrs = append(allErrs, validation.NewInvalidError("path", downwardAPIVolumeFile.Path, "must not contain \"..\"."))
-			}
-		}
-		if strings.HasPrefix(items[0], "..") && len(items[0]) > 2 {
-			allErrs = append(allErrs, validation.NewInvalidError("path", downwardAPIVolumeFile.Path, "must not start with \"..\"."))
-		}
+		allErrs = append(allErrs, validateVolumeSourcePath(downwardAPIVolumeFile.Path, "path")...)
 		allErrs = append(allErrs, validateObjectFieldSelector(&downwardAPIVolumeFile.FieldRef, &validDownwardAPIFieldPathExpressions).Prefix("FieldRef")...)
 	}
 	return allErrs
 }
 
+// This validate will make sure targetPath:
+// 1. is not abs path
+// 2. does not contain '..' or ' '
+// 3. does not start with '..'
+func validateVolumeSourcePath(targetPath string, field string) validation.ErrorList {
+	allErrs := validation.ErrorList{}
+	if path.IsAbs(targetPath) {
+		allErrs = append(allErrs, validation.NewForbiddenError(field, "must not be an absolute path"))
+	}
+	items := strings.Split(targetPath, string(os.PathSeparator))
+	for index, item := range items {
+		item = strings.TrimSpace(item)
+		if item == ".." {
+			allErrs = append(allErrs, validation.NewInvalidError(field, targetPath, "must not contain \"..\""))
+		}
+		if index != 0 && len(item) == 0 {
```

Comment:
I don't think this does what it's supposed to.  Why is index 0 special-cased here?  Also, I'm pretty sure this will let some spaces through -- for example `"foo bar"` would be allowed by this, which is contrary to the error message.  If you want to test for spaces, you have to test for the index of a space (using `[\s\xA0]`, which will also capture nonbreaking spaces) in the trimmed string.  However, I don't know that we want to disallow spaces in names.  Since this is a new condition, maybe we ought to break the question of how to handle spaces out into a different issue.